Source/TCPConnection.swift

```diff
@@ -137,32 +137,28 @@ public final class TCPConnection: Connection {
             tcprecvuntil(socket, $0.baseAddress, $0.count, delimiter, delimiter.utf8.count, deadline.int64milliseconds)
         }
 
-        try TCPError.assertNoReceiveErrorWithData(data, bytesProcessed: bytesProcessed)
+        try TCPError.assertNoReceiveError(withData: data, bytesProcessed: bytesProcessed)
         return Data(data.prefix(bytesProcessed))
     }
 
-    public func close() -> Bool {
+    public func close() throws {
         guard let socket = self.socket else {
             closed = true
-            return true
+            throw TCPError.closedSocket(description: "Connection has not been initialized. You must first open to the connection.")
```

Comment:
@paulofaria in earlier versions, if `socket` is nil the function would just return `true` and mark connection as `closed`. Should we throw an error in this situation, or stick with previous behaviour?